src/content/blog/astro-4140.mdx

```diff
@@ -0,0 +1,271 @@
+coverImage: "/src/content/blog/_images/astro-4140/header_astro_4_14.webp"
+socialImage: "/src/content/blog/_images/astro-4140/og_image_4_14.webp"
+lang: "en"
+---
+
+import BlogContentImage from "/src/components/BlogContentImage.astro";
+import BlogSummary from "/src/components/BlogSummary.astro";
+import frontmatterIntellisense from "/src/content/blog/_images/astro-4140/frontmatter-intellisense.png";
+
+<BlogSummary>
+  Astro 4.14 is out now! This release includes the first experimental release of the Content Layer API, our new super-flexible solution for managing content in Astro projects, support for Intellisense inside content files, and more.
+</BlogSummary>
+
+This release includes the following highlights:
+
+- [**Experimental: Content Layer API**](#experimental-content-layer-api)
+- [**Experimental: Intellisense inside content files**](#experimental-intellisense-inside-content-files)
+- [**Deprecate support for dynamic `prerender` values**](#deprecate-support-for-dynamic-prerender-values)
+- [**New `injectTypes` integration API**](#new-injecttypes-integration-api)
+- [**Support for metastrings in the Code component**](#support-for-metastrings-in-the-code-component)
+
+To upgrade an existing project, use the automated `@astrojs/upgrade` CLI tool. Alternatively, upgrade manually by running the upgrade command for your package manager:
+
+```sh
+# Recommended:
+npx @astrojs/upgrade
+
+# Manual:
+npm install astro@latest
+pnpm upgrade astro --latest
+yarn upgrade astro --latest
+```
+
+## Experimental: Content Layer API
+
+In Astro 2.0 we introduced [Content Collections](https://docs.astro.build/en/guides/content-collections/), a powerful way to manage local content in Astro projects. While this has helped hundreds of thousands of developers build content-driven sites with Astro, we've heard your feedback that you want more flexibility and power in managing your content. We're excited to announce the experimental release of the next generation of content collections in Astro: the Content Layer API.
+
+The new Content Layer API builds upon content collections, taking them beyond local files in `src/content/` and allowing you to fetch content from anywhere, including remote APIs. These new collections work alongside your existing content collections, and you can migrate them to the new API at your own pace. We think you'll want to soon, because they have lots of advantages, but there's no rush – the existing collections will continue to work as they always have.
+
+
+### Performance
+
+The Content Layer API is the first step in scaling Astro to support fast builds for sites with tens or hundreds of thousands of pages. The new API is designed to be more efficient and performant in the way it loads content. It caches content locally to avoid the need to keep hitting APIs, and has dramatically improved handling of local Markdown and MDX files. This benchmark shows the performance when building a site with 10,000 pages on a Macbook Air M1:
```

Comment:
unnecessary comma
```suggestion
The Content Layer API is the first step in scaling Astro to support fast builds for sites with tens or hundreds of thousands of pages. The new API is designed to be more efficient and performant in the way it loads content. It caches content locally to avoid the need to keep hitting APIs and has dramatically improved handling of local Markdown and MDX files. This benchmark shows the performance when building a site with 10,000 pages on a Macbook Air M1:
```